src/DDTrace/Integrations/OpenAI/OpenAIIntegration.php

```diff
@@ -0,0 +1,1220 @@
+                        );
+                    }
+                ]
+            );
+        }
+
+        foreach ($streamedTargets as [$class, $method, $methodName, $httpMethod, $endpoint]) {
+            \DDTrace\trace_method(
+                $class,
+                $method,
+                [
+                    'prehook' => function (\DDTrace\SpanData $span, $args) use ($methodName, $integration) {
+                        $integration->setServiceName($span);
+                        $clientData = ObjectKVStore::get($this, 'client_data');
+                        if (\is_null($clientData)) {
+                            $transporter = ObjectKVStore::get($this, 'transporter');
+                            $clientData = ObjectKVStore::get($transporter, 'client_data');
+                            ObjectKVStore::put($this, 'client_data', $clientData);
+                        }
+                        /** @var array{baseUri: string, headers: string, apiKey: ?string} $clientData */
+                        OpenAIIntegration::handleRequest(
+                            span: $span,
+                            methodName: $methodName,
+                            args: $args,
+                            basePath: $clientData['baseUri'],
+                            apiKey: $clientData['apiKey'],
+                            streamed: true
+                        );
+                    },
+                    'posthook' => function (\DDTrace\SpanData $span, $args, $response) use ($logger, $httpMethod, $endpoint) {
+                        /** @var \OpenAI\Responses\StreamResponse $response */
+                        OpenAIIntegration::handleStreamedResponse(
+                            span: $span,
+                            logger: $logger,
+                            headers: $response->meta()->toArray(),
+                            response: $response,
+                            httpMethod: $httpMethod,
+                            endpoint: $endpoint,
+                        );
+                    }
+                ]
+            );
+        }
+
+        \DDTrace\install_hook(
+            'OpenAI\Responses\StreamResponse::getIterator',
+            function (HookData $hook) {
+                /** @var \OpenAI\Responses\StreamResponse $this */
+                $generatorClosure = ObjectKVStore::get($this, 'generator');
+                if (!is_null($generatorClosure)) {
+                    // It is valid for the retval to be empty if the generator was already consumed
+                    $hook->overrideReturnValue($generatorClosure());
+                }
+            }
+        );
+
+        return Integration::LOADED;
+    }
+
+    public static function normalizeRequestPayload(
+        string $methodName,
+        array  $args
+    ): array
+    {
+        switch ($methodName) {
+            case 'listModels':
+            case 'listFiles':
+            case 'listFineTunes':
```

Comment:
nit: What do you think about creating some constants for these given that they are used in a bunch of places?